### PR TITLE
fix: add missing package result

### PR DESCRIPTION
**Pull request by gene-redpanda**
Removed during the change to add airgap. Allows package changes to trigger restart.

### CODE REVIEW

**Comment by tmgstevens** [comment]
@gene-redpanda I've managed to squash the three `package` statements into one by careful use of `default` meaning that we can still get the desired behaviour of `allow_unauthenticated` and also the `http[s]_proxy` variables.

Tested for Debian and appears to be working fine now (it wasn't before I made the changes)

**Comment by tmgstevens** [comment]
Reverting my commits due to poor branch etiquette - please accept my apologies

**Comment by tmgstevens** [comment]
Tested the fix as it stands and it does not work for debian packages:
```
TASK [redpanda.cluster.redpanda_broker : Install redpanda from repository (Proxy)] **************************************************************************************************************************************************************
skipping: [52.35.135.46] => {"changed": false, "skip_reason": "Conditional result was False"}
skipping: [52.41.230.210] => {"changed": false, "skip_reason": "Conditional result was False"}
skipping: [34.212.187.105] => {"changed": false, "skip_reason": "Conditional result was False"}

TASK [redpanda.cluster.redpanda_broker : Install redpanda from repository (Standard)] ***********************************************************************************************************************************************************
changed: [52.41.230.210] => {"cache_update_time": 1719221815, "cache_updated": true, "changed": true, "stderr": "", "stderr_lines": [], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nThe following packages will be upgraded:\n  redpanda\n1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.\nNeed to get 48.6 MB of archives.\nAfter this operation, 694 kB of additional disk space will be used.\nGet:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]\nFetched 48.6 MB in 3s (14.6 MB/s)\n(Reading database ... \r(Reading database ... 5%\r(Reading database ... 10%\r(Reading database ... 15%\r(Reading database ... 20%\r(Reading database ... 25%\r(Reading database ... 30%\r(Reading database ... 35%\r(Reading database ... 40%\r(Reading database ... 45%\r(Reading database ... 50%\r(Reading database ... 55%\r(Reading database ... 60%\r(Reading database ... 65%\r(Reading database ... 70%\r(Reading database ... 75%\r(Reading database ... 80%\r(Reading database ... 85%\r(Reading database ... 90%\r(Reading database ... 95%\r(Reading database ... 100%\r(Reading database ... 62215 files and directories currently installed.)\r\nPreparing to unpack .../redpanda_23.3.10-1_amd64.deb ...\r\nUnpacking redpanda (23.3.10-1) over (23.3.9-1) ...\r\nSetting up redpanda (23.3.10-1) ...\r\nredpanda:x:997:998::/var/lib/redpanda:/bin/sh\r\nredpanda:x:998:\r\n\r\nTo get the most out of the fastest queue in the west, enable production mode by\r\nrunning the following:\r\n\r\n    sudo rpk redpanda mode production\r\n\r\nfollowed by:\r\n\r\n    sudo rpk redpanda tune all\r\n    sudo systemctl start redpanda\r\n\r\nThis will autotune your system to give you the best performance from Redpanda.\r\nYou can get more information on the tuning parameters here:\r\nhttps://docs.redpanda.com/docs/introduction/autotune/\r\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "The following packages will be upgraded:", "  redpanda", "1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.", "Need to get 48.6 MB of archives.", "After this operation, 694 kB of additional disk space will be used.", "Get:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]", "Fetched 48.6 MB in 3s (14.6 MB/s)", "(Reading database ... ", "(Reading database ... 5%", "(Reading database ... 10%", "(Reading database ... 15%", "(Reading database ... 20%", "(Reading database ... 25%", "(Reading database ... 30%", "(Reading database ... 35%", "(Reading database ... 40%", "(Reading database ... 45%", "(Reading database ... 50%", "(Reading database ... 55%", "(Reading database ... 60%", "(Reading database ... 65%", "(Reading database ... 70%", "(Reading database ... 75%", "(Reading database ... 80%", "(Reading database ... 85%", "(Reading database ... 90%", "(Reading database ... 95%", "(Reading database ... 100%", "(Reading database ... 62215 files and directories currently installed.)", "Preparing to unpack .../redpanda_23.3.10-1_amd64.deb ...", "Unpacking redpanda (23.3.10-1) over (23.3.9-1) ...", "Setting up redpanda (23.3.10-1) ...", "redpanda:x:997:998::/var/lib/redpanda:/bin/sh", "redpanda:x:998:", "", "To get the most out of the fastest queue in the west, enable production mode by", "running the following:", "", "    sudo rpk redpanda mode production", "", "followed by:", "", "    sudo rpk redpanda tune all", "    sudo systemctl start redpanda", "", "This will autotune your system to give you the best performance from Redpanda.", "You can get more information on the tuning parameters here:", "https://docs.redpanda.com/docs/introduction/autotune/"]}
changed: [52.35.135.46] => {"cache_update_time": 1719221815, "cache_updated": true, "changed": true, "stderr": "", "stderr_lines": [], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nThe following packages will be upgraded:\n  redpanda\n1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.\nNeed to get 48.6 MB of archives.\nAfter this operation, 694 kB of additional disk space will be used.\nGet:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]\nFetched 48.6 MB in 3s (16.4 MB/s)\n(Reading database ... \r(Reading database ... 5%\r(Reading database ... 10%\r(Reading database ... 15%\r(Reading database ... 20%\r(Reading database ... 25%\r(Reading database ... 30%\r(Reading database ... 35%\r(Reading database ... 40%\r(Reading database ... 45%\r(Reading database ... 50%\r(Reading database ... 55%\r(Reading database ... 60%\r(Reading database ... 65%\r(Reading database ... 70%\r(Reading database ... 75%\r(Reading database ... 80%\r(Reading database ... 85%\r(Reading database ... 90%\r(Reading database ... 95%\r(Reading database ... 100%\r(Reading database ... 62215 files and directories currently installed.)\r\nPreparing to unpack .../redpanda_23.3.10-1_amd64.deb ...\r\nUnpacking redpanda (23.3.10-1) over (23.3.9-1) ...\r\nSetting up redpanda (23.3.10-1) ...\r\nredpanda:x:997:998::/var/lib/redpanda:/bin/sh\r\nredpanda:x:998:\r\n\r\nTo get the most out of the fastest queue in the west, enable production mode by\r\nrunning the following:\r\n\r\n    sudo rpk redpanda mode production\r\n\r\nfollowed by:\r\n\r\n    sudo rpk redpanda tune all\r\n    sudo systemctl start redpanda\r\n\r\nThis will autotune your system to give you the best performance from Redpanda.\r\nYou can get more information on the tuning parameters here:\r\nhttps://docs.redpanda.com/docs/introduction/autotune/\r\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "The following packages will be upgraded:", "  redpanda", "1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.", "Need to get 48.6 MB of archives.", "After this operation, 694 kB of additional disk space will be used.", "Get:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]", "Fetched 48.6 MB in 3s (16.4 MB/s)", "(Reading database ... ", "(Reading database ... 5%", "(Reading database ... 10%", "(Reading database ... 15%", "(Reading database ... 20%", "(Reading database ... 25%", "(Reading database ... 30%", "(Reading database ... 35%", "(Reading database ... 40%", "(Reading database ... 45%", "(Reading database ... 50%", "(Reading database ... 55%", "(Reading database ... 60%", "(Reading database ... 65%", "(Reading database ... 70%", "(Reading database ... 75%", "(Reading database ... 80%", "(Reading database ... 85%", "(Reading database ... 90%", "(Reading database ... 95%", "(Reading database ... 100%", "(Reading database ... 62215 files and directories currently installed.)", "Preparing to unpack .../redpanda_23.3.10-1_amd64.deb ...", "Unpacking redpanda (23.3.10-1) over (23.3.9-1) ...", "Setting up redpanda (23.3.10-1) ...", "redpanda:x:997:998::/var/lib/redpanda:/bin/sh", "redpanda:x:998:", "", "To get the most out of the fastest queue in the west, enable production mode by", "running the following:", "", "    sudo rpk redpanda mode production", "", "followed by:", "", "    sudo rpk redpanda tune all", "    sudo systemctl start redpanda", "", "This will autotune your system to give you the best performance from Redpanda.", "You can get more information on the tuning parameters here:", "https://docs.redpanda.com/docs/introduction/autotune/"]}
changed: [34.212.187.105] => {"cache_update_time": 1719221815, "cache_updated": true, "changed": true, "stderr": "", "stderr_lines": [], "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nThe following packages will be upgraded:\n  redpanda\n1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.\nNeed to get 48.6 MB of archives.\nAfter this operation, 694 kB of additional disk space will be used.\nGet:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]\nFetched 48.6 MB in 3s (15.9 MB/s)\n(Reading database ... \r(Reading database ... 5%\r(Reading database ... 10%\r(Reading database ... 15%\r(Reading database ... 20%\r(Reading database ... 25%\r(Reading database ... 30%\r(Reading database ... 35%\r(Reading database ... 40%\r(Reading database ... 45%\r(Reading database ... 50%\r(Reading database ... 55%\r(Reading database ... 60%\r(Reading database ... 65%\r(Reading database ... 70%\r(Reading database ... 75%\r(Reading database ... 80%\r(Reading database ... 85%\r(Reading database ... 90%\r(Reading database ... 95%\r(Reading database ... 100%\r(Reading database ... 62215 files and directories currently installed.)\r\nPreparing to unpack .../redpanda_23.3.10-1_amd64.deb ...\r\nUnpacking redpanda (23.3.10-1) over (23.3.9-1) ...\r\nSetting up redpanda (23.3.10-1) ...\r\nredpanda:x:997:998::/var/lib/redpanda:/bin/sh\r\nredpanda:x:998:\r\n\r\nTo get the most out of the fastest queue in the west, enable production mode by\r\nrunning the following:\r\n\r\n    sudo rpk redpanda mode production\r\n\r\nfollowed by:\r\n\r\n    sudo rpk redpanda tune all\r\n    sudo systemctl start redpanda\r\n\r\nThis will autotune your system to give you the best performance from Redpanda.\r\nYou can get more information on the tuning parameters here:\r\nhttps://docs.redpanda.com/docs/introduction/autotune/\r\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information...", "The following packages will be upgraded:", "  redpanda", "1 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.", "Need to get 48.6 MB of archives.", "After this operation, 694 kB of additional disk space will be used.", "Get:1 https://dl.redpanda.com/public/redpanda/deb/ubuntu focal/main amd64 redpanda amd64 23.3.10-1 [48.6 MB]", "Fetched 48.6 MB in 3s (15.9 MB/s)", "(Reading database ... ", "(Reading database ... 5%", "(Reading database ... 10%", "(Reading database ... 15%", "(Reading database ... 20%", "(Reading database ... 25%", "(Reading database ... 30%", "(Reading database ... 35%", "(Reading database ... 40%", "(Reading database ... 45%", "(Reading database ... 50%", "(Reading database ... 55%", "(Reading database ... 60%", "(Reading database ... 65%", "(Reading database ... 70%", "(Reading database ... 75%", "(Reading database ... 80%", "(Reading database ... 85%", "(Reading database ... 90%", "(Reading database ... 95%", "(Reading database ... 100%", "(Reading database ... 62215 files and directories currently installed.)", "Preparing to unpack .../redpanda_23.3.10-1_amd64.deb ...", "Unpacking redpanda (23.3.10-1) over (23.3.9-1) ...", "Setting up redpanda (23.3.10-1) ...", "redpanda:x:997:998::/var/lib/redpanda:/bin/sh", "redpanda:x:998:", "", "To get the most out of the fastest queue in the west, enable production mode by", "running the following:", "", "    sudo rpk redpanda mode production", "", "followed by:", "", "    sudo rpk redpanda tune all", "    sudo systemctl start redpanda", "", "This will autotune your system to give you the best performance from Redpanda.", "You can get more information on the tuning parameters here:", "https://docs.redpanda.com/docs/introduction/autotune/"]}

TASK [redpanda.cluster.redpanda_broker : Install redpanda from repository (GCP)] ****************************************************************************************************************************************************************
skipping: [52.35.135.46] => {"changed": false, "skip_reason": "Conditional result was False"}
skipping: [52.41.230.210] => {"changed": false, "skip_reason": "Conditional result was False"}
skipping: [34.212.187.105] => {"changed": false, "skip_reason": "Conditional result was False"}

<snip>

TASK [redpanda.cluster.redpanda_broker : Debug] *************************************************************************************************************************************************************************************************
ok: [52.35.135.46] => {
    "package_result": {
        "changed": false,
        "skip_reason": "Conditional result was False",
        "skipped": true
    }
}
ok: [52.41.230.210] => {
    "package_result": {
        "changed": false,
        "skip_reason": "Conditional result was False",
        "skipped": true
    }
}
ok: [34.212.187.105] => {
    "package_result": {
        "changed": false,
        "skip_reason": "Conditional result was False",
        "skipped": true
    }
}

TASK [redpanda.cluster.redpanda_broker : Establish whether restart required] ********************************************************************************************************************************************************************
ok: [52.35.135.46] => {"ansible_facts": {"restart_required": false}, "changed": false}
ok: [52.41.230.210] => {"ansible_facts": {"restart_required": false}, "changed": false}
ok: [34.212.187.105] => {"ansible_facts": {"restart_required": false}, "changed": false}
```

The reason for this is that `register` still registers a result even if the `when` clause evaluates to `false`.
So we either need to do some logic on the results (example here: https://stackoverflow.com/a/34622173) or I've got a fix that will work which folds down the number of `package` statements down to one. 

I'm happy with either approach, I'll raise a PR for mine but happy for you to amend this one.

**Comment by tmgstevens** [comment]
Should also add that we need to `register` the result of the task on line 11.

File: roles/redpanda_broker/tasks/start-redpanda.yml [hunk 1]
```diff
@@ -35,17 +35,12 @@
     verbosity: 3
   when: redpanda is defined
 
-- name: Debug
-  ansible.builtin.debug:
-    msg: "{{ redpanda | from_json }}"
-    verbosity: 3
-  when: redpanda is defined
-
```
Comment (tmgstevens) [inline roles/redpanda_broker/tasks/start-redpanda.yml:43]: Why removing this? Debugging when `redpanda` goes wrong is really hard, so this helps at verbosity 3.

Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/start-redpanda.yml:43]: We already have another one that is just the string, seems unnecessary to duplicate it

Comment (gene-redpanda) [reply roles/redpanda_broker/tasks/start-redpanda.yml]: I added it back 🤷‍♀️

Comment (tmgstevens) [reply roles/redpanda_broker/tasks/start-redpanda.yml]: The reason, for posterity, is that if the JSON doesn't parse, then it becomes obvious at this point. So the first one dumps the string, the second one dumps the JSON. Possibly not needed I guess, but if it fails on this line then you know why

**Pull request closed without merging**